lib/src/configuration.dart

```diff
@@ -70,65 +70,197 @@ class Configuration {
     return Directory.current.absolute.path;
   }
 
-  /// The schema version used to open the [Realm]
+  /// Specifies the FIFO special files fallback location.
   ///
-  /// If omitted the default value of `0` is used to open the [Realm]
-  /// It is required to specify a schema version when initializing an existing
-  /// Realm with a schema that contains objects that differ from their previous
-  /// specification. If the schema was updated and the schemaVersion was not,
-  /// an [RealmException] will be thrown.
-  final int schemaVersion;
+  /// Opening a [Realm] creates a number of FIFO special files in order to
+  /// coordinate access to the [Realm] across threads and processes. If the [Realm] file is stored in a location
+  /// that does not allow the creation of FIFO special files (e.g. FAT32 filesystems), then the [Realm] cannot be opened.
+  /// In that case [Realm] needs a different location to store these files and this property defines that location.
+  /// The FIFO special files are very lightweight and the main [Realm] file will still be stored in the location defined
+  /// by the [path] you  property. This property is ignored if the directory defined by [path] allow FIFO special files.
+  String? get fifoFilesFallbackPath;
 
   /// The path where the Realm should be stored.
   ///
   /// If omitted the [defaultPath] for the platform will be used.
+  String get path;
+
+  /// The [RealmSchema] for this [Configuration]
+  RealmSchema get schema;
+  List<int>? get encryptionKey;
+
+  @Deprecated('Use Configuration.local instead')
+  factory Configuration(
+    List<SchemaObject> schemaObjects, {
+    InitialDataCallback? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+    bool disableFormatUpgrade,
+    bool isReadOnly,
+    ShouldCompactCallback? shouldCompactCallback,
+  }) = LocalConfiguration;
+
+  factory Configuration.local(
+    List<SchemaObject> schemaObjects, {
+    InitialDataCallback? initialDataCallback,
+    int schemaVersion,
+    String? fifoFilesFallbackPath,
+    String? path,
+    bool disableFormatUpgrade,
+    bool isReadOnly,
+    ShouldCompactCallback? shouldCompactCallback,
+  }) = LocalConfiguration;
+
+  factory Configuration.inMemory(
```

Comment:
Why not having initialDataCallback fon InMemory?

Reply:
Because in memory databases are always created from scratch, instead of using `initialDataCallback` just always do whatever you wanted to in there directly on the realm on startup.